### ISSUE TITLE
AOT build fails with tsconfig.json paths

### ISSUE THREAD

**Issue by bialad** [angular/angular-cli#2977]
### OS?
Windows 10

### Versions.
angular-cli: 1.0.0-beta.19-3
node: 6.9.1
os: win32 x64

### Repro
**In tsconfig.json:**
```
"baseUrl": ".",
"paths": {
  "@shared/*": [
    "app/shared/*"
  ],
},
```
**In component:**
```
import { url } from '@shared/settings';
```
**I then run:**
```
ng build -prod -aot
```

### The log given by the failure.
```
26% building modules 135/212 modules 77 active ...node_modules\@angular\common\index.jsresolve '@shared/settings' in 'D:\projects\git\brfportal.se\app\src\app\services'
  Parsed request is a module
  using description file: D:\projects\git\brfportal.se\app\package.json (relative path: ./src/app/services)
    Field 'browser' doesn't contain a valid alias configuration
  after using description file: D:\projects\git\brfportal.se\app\package.json (relative path: ./src/app/services)
    resolve as module
      looking for modules in D:\projects\git\brfportal.se\app\node_modules
        using description file: D:\projects\git\brfportal.se\app\package.json (relative path: ./node_modules)
          Field 'browser' doesn't contain a valid alias configuration
        after using description file: D:\projects\git\brfportal.se\app\package.json (relative path: ./node_modules)
          using description file: D:\projects\git\brfportal.se\app\package.json (relative path: ./node_modules/@shared/settings)
            as directory
              D:\projects\git\brfportal.se\app\node_modules\@shared\settings doesn't exist
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              D:\projects\git\brfportal.se\app\node_modules\@shared\settings doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              D:\projects\git\brfportal.se\app\node_modules\@shared\settings.ts doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              D:\projects\git\brfportal.se\app\node_modules\@shared\settings.js doesn't exist

D:\projects\git\brfportal.se\app\node_modules\webpack\lib\Compiler.js:220
                                        if(err) return callback(err);
                    ^
TypeError: Cannot read property 'hash' of undefined
    at D:\projects\git\brfportal.se\app\node_modules\angular-cli\tasks\build-webpack.js:35:26
    at Compiler.onCompiled (D:\projects\git\brfportal.se\app\node_modules\webpack\lib\Compiler.js:220:21)
    at Compiler.<anonymous> (D:\projects\git\brfportal.se\app\node_modules\webpack\lib\Compiler.js:464:19)
    at D:\projects\git\brfportal.se\app\node_modules\tapable\lib\Tapable.js:148:11
    at Compilation.<anonymous> (D:\projects\git\brfportal.se\app\node_modules\webpack\lib\Compilation.js:443:11)
...
```

### Mention any other details that might be useful.

If I add this field to package.json, it works. But just for that specific file.
```
"browser": {
  "@shared/settings": "./src/app/shared/settings"
},
```
I use this aliasing successfully over hundreds of files using `ng serve`, `ng build` and `ng build -prod`.

Somehow the cli should handle this aliasing for AOT right?

**Comment by psurrey** [comment]
@bialad We are also using aliasing with the `"paths"` setting in tsconfig.json. Did it just work for you (apart from aot)? Did you just add the paths to tsconfig or is it required to adapt something for webpack as well?

**Comment by bialad** [comment]
It Linda "just" worked. :)

There was a bug in web pack (may be fixed) where you couldn't use the pathing where the caller is a sibling folder of the target, and a sub folder of the path.

**Comment by bialad** [comment]
This has been tagged for a while now,. But could someone confirm this as a bug, and maybe comment on the severity and how difficult a fix would be. It's it not possible to use aliasing and aot together?

**Comment by bialad** [comment]
After the latest beta-22 release, I'm getting the same errors when just running "ng serve".

I've tried to debug this, and see if there's something I can do. But the filename and path parsing is way over my head. This is forcing me (unless I stay at beta-21) to abandon absolute paths all together...

I don't see this issue mentioned anywhere else. I'm I the only one experiencing this, our the only one using absolute paths in my project. :(

**Comment by rolandoldengarm** [comment]
I have got the same issue, but just with specifying rootDir, not using aliases. But then e.g. import... from 'app/domain' fails.

**Comment by skovmand** [comment]
I have it too. No idea why.

**Comment by leroyvaughan** [comment]
I managed to work around this for "npm start" and "ng serve" when having problems with relative paths.
I converted a "systemjs angular2" app to an "angular cli" app.  
In my code I had to do this in my files (for this example I used "./app/dashboard/dashboard.component.ts":

templateUrl: 'dashboard.component.html'  >> '../dashboard/dashboard.component.html'

![screenhunter_193 dec 24 16 43](https://cloud.githubusercontent.com/assets/5483798/21469291/5fab8880-c9f8-11e6-80ca-14ba0aa168ec.jpg)

What I did was traverse back up to the parent folder, then go right back down into the folder (./app/dashboard) that my dashboard.component.ts file was in.  *Notice that I did the same for my references to the Hero imports on lines 3 and 4 as well.


I tried all other path types and this works both ways with _npm start_ and _ng serve path resolution errors for me.
